### ISSUE TITLE
[core] xml output doesn't escape CDATA inside its own CDATA

### ISSUE THREAD

**Issue by stephen-carter-at-sf** [pmd/pmd#5059]
**Affects PMD Version:** 

Make sure, to test with the latest PMD version.

**Description:**

Often, when an apex file contains CDATA, then if an error occurs while parsing this file with PMD... then it attempts to put the file contents in its own CDATA tag... but fails to escape the CDATA from the file.

So you end up with something like:
```
OPEN                 OPEN                 CLOSE            CLOSE
<![CDATA[    ....    <![CDATA[    ....    ]]>          ... ]]>
```
which is interpretted by xml readers as

```
OPEN                 IGNORED              CLOSE            ?WHAT THE?
<![CDATA[    ....    <![CDATA[    ....    ]]>          ... ]]>
```

**PMD Version:**

```
PMD 7.2.0 (38bc997d4e7f4b29cee86829915f17f925211800, 2024-05-31T10:33:57Z)
Java version: 11.0.17.0.1, vendor: Azul Systems, Inc., runtime: /Users/stephen.carter/blt/tools/Darwin/jdk/openjdk_11.0.17.0.1_11.60.54_x64

```

**Steps to reproduce:**

Please provide detailed steps for how we can reproduce the bug.

* Download the [reproFiles.zip](https://github.com/user-attachments/files/15745360/reproFiles.zip) file that is attached and unzip it.
* Then run:
```
pmd check -d dummy.cls -R my_ruleset.xml --format xml > output.xml
```
* This produces corrupted (invalid) xml output that can't be read... thus crashing our pipeline.

Any xml reader will crash... but we are reading the xml from within a node application. And so you can easily reproduce what we see with:
```
~/temp: npm install xml-js

~/temp: node              
Welcome to Node.js v21.7.1.

> const { xml2js } = await import("xml-js");
undefined
> const xmlData = fs.readFileSync('./output.xml','utf-8');
undefined
> const pmdData = xml2js(xmlData, {compact: false, ignoreDeclaration: true});
Uncaught:
<ref *1> Error: Unexpected close tag
Line: 158
Column: 8
Char: >
```
**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*

CLI

**Comment by adangel** [comment]
@stephen-carter-at-sf Can you give more information about this specific file? Is it supposed to be parsed correctly (then we would have another issue)? The file doesn't look like a valid Apex source file at all - how is it used in real world scenario? Thanks!

**Comment by jsotuyod** [comment]
@stephen-carter-at-sf just FYI, the escaping has been fixed in PMD 7.6.0

**Comment by stephen-carter-at-sf** [comment]
Oh great. I assumed it was just closed. Thanks.
